src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -877,6 +880,9 @@ else if (conf.commitlog_segment_size.toMebibytes() >= 2048)
         else if (conf.commitlog_segment_size.toKibibytes() < 2 * conf.max_mutation_size.toKibibytes())
             throw new ConfigurationException("commitlog_segment_size must be at least twice the size of max_mutation_size / 1024", false);
 
+        if (conf.crypto_provider != null)
```

Comment:
The initialization logic should be rewritten, try to look at `applySeedProvider();` how it is done to take an inspiration from there.